corehq/apps/case_search/temp/fix_child_cases.py

```diff
@@ -0,0 +1,85 @@
+                # --------------
+
+                child_case_helper = CaseHelper(domain=DOMAIN, case=child_case)
+                
+                """
+                    Step 2 - Make sure that the case properties of both the child and parent cases correspond to te
+                    village's hierarchy. The owner_id of both cases is the village id
+                """
+                try:
+                    parent_case = child_case.parent
+                    village_id = parent_case.owner_id
+                except Exception:
+                    logfile.write(f"Skipped {child_case.case_id}. Reason: No parent case\n")
+                    continue
+
+                try:
+                    village = SQLLocation.objects.filter(location_id=village_id, domain=DOMAIN).first()
+                    formation_sanitaire = village.parent
+                    arrondissement = formation_sanitaire.parent
+                    commune = arrondissement.parent
+                    zone_sanitaire = commune.parent
+                    departement = zone_sanitaire.parent
+                except Exception as e:
+                    logfile.write(f"Skipped {child_case.case_id}. Reason: Location hierarchy issue. {e}\n")
+                    continue
+
+                correct_properties = {
+                    "hh_village_name": village.name,
+                    "hh_formation_sanitaire_name": formation_sanitaire.name,
+                    "hh_arrondissement_name": arrondissement.name,
+                    "hh_commune_name": commune.name,
+                    "hh_zone_sanitaire_name": zone_sanitaire.name,
+                    "hh_departement_name": departement.name
+                }
+
+                parent_case_helper = CaseHelper(domain=DOMAIN, case=parent_case)
+                case_helpers = (parent_case_helper, child_case_helper)
+                cases = (parent_case, child_case)
+
+                try:
+                    for case, case_helper in zip(cases, case_helpers):
+                        is_dirty = False
+                        updated_properties = {}
+                        for property_name, correct_property_value in correct_properties.items():
+                            curr_case_property_value = case.get_case_property(property_name)
+
+                            if (not curr_case_property_value) or (curr_case_property_value != correct_property_value):
+                                updated_properties[property_name] = correct_property_value
+                                is_dirty = True
+                        
+                        update_dict = {'properties': updated_properties}
+                        if case.case_id == child_case.case_id and case.owner_id != parent_case.owner_id:
+                            is_dirty = True
+                            update_dict['owner_id'] = parent_case.owner_id
+
+                        if is_dirty:
+                            case_helper.update(update_dict)
```

Comment:
This doesn't do updates in bulk, does it? You could consider submit_case_blocks that can submit chunks of cases.